ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template5.kt

```diff
@@ -0,0 +1,453 @@
+) {
+    Column {
+        var packageSelectorVisible by remember {
+            mutableStateOf(state.templateConfiguration.mode != PaywallMode.FOOTER_CONDENSED)
+        }
+
+        if (state.isInFullScreenMode) {
+            HeaderImage(state.templateConfiguration.images.headerUri)
+        }
+
+        Template5MainContent(state, viewModel, packageSelectorVisible)
+
+        AnimatedVisibility(
+            visible = packageSelectorVisible,
+            enter = fadeIn(animationSpec = UIConstant.defaultAnimation()),
+            exit = fadeOut(animationSpec = UIConstant.defaultAnimation()),
+            label = "Template2.packageSpacing",
+        ) {
+            Spacer(modifier = Modifier.height(UIConstant.defaultVerticalSpacing))
+        }
+
+        PurchaseButton(state, viewModel)
+
+        Footer(
+            templateConfiguration = state.templateConfiguration,
+            viewModel = viewModel,
+            allPlansTapped = { packageSelectorVisible = !packageSelectorVisible },
+        )
+    }
+}
+
+@Suppress("LongMethod")
+@Composable
+private fun ColumnScope.Template5MainContent(
+    state: PaywallState.Loaded,
+    viewModel: PaywallViewModel,
+    packageSelectionVisible: Boolean,
+) {
+    Column(
+        modifier = Modifier
+            .verticalScroll(rememberScrollState())
+            .conditional(state.isInFullScreenMode) {
+                Modifier.weight(1f)
+            }
+            .padding(horizontal = UIConstant.defaultHorizontalPadding, vertical = UIConstant.defaultVerticalSpacing),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.spacedBy(UIConstant.defaultVerticalSpacing, Alignment.CenterVertically),
+    ) {
+        if (state.isInFullScreenMode) {
+//            Spacer(Modifier.weight(1f))
+
+            val localizedConfig = state.selectedLocalization
+            val colors = state.templateConfiguration.getCurrentColors()
+            Markdown(
+                style = MaterialTheme.typography.displaySmall,
+                fontWeight = FontWeight.Bold,
+                textAlign = TextAlign.Left,
+                text = localizedConfig.title,
+                color = colors.text1,
+            )
+            Spacer(Modifier.weight(1f))
+
+            Features(
+                features = localizedConfig.features,
+                colors = colors,
+            )
+
+            Spacer(Modifier.weight(1f))
+        }
+
+        AnimatedPackages(state, packageSelectionVisible, viewModel)
+
+        if (state.isInFullScreenMode) {
+            Spacer(Modifier.weight(1f))
+        }
+    }
+}
+
+@Composable
+private fun HeaderImage(uri: Uri?) {
+    uri?.let {
+        RemoteImage(
+            urlString = uri.toString(),
+            modifier = Modifier
+                .aspectRatio(ratio = Template5UIConstants.headerAspectRatio),
+            contentScale = ContentScale.Crop,
+        )
+    }
+}
+
+@Composable
+private fun Features(
```

Comment:
I was wondering whether we could share this between templates 3 and 5... But comparing the code, it would be tricky I think, so we can leave it separated.

Reply:
Yeah plus it's slightly different.